system/HTTP/IncomingRequest.php

```diff
@@ -568,10 +568,18 @@ public function getVar($index = null, $filter = null, $flags = null)
      * @see http://php.net/manual/en/function.json-decode.php
      *
      * @return array|bool|float|int|stdClass|null
+     *
+     * @throws HTTPException When the body is invalid as JSON.
      */
     public function getJSON(bool $assoc = false, int $depth = 512, int $options = 0)
     {
-        return json_decode($this->body ?? '', $assoc, $depth, $options);
+        $result = json_decode($this->body ?? '', $assoc, $depth, $options);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
```

Comment:
I think we need to check for `JSON_THROW_ON_ERROR`. From the docs:
> Returns the last error (if any) occurred during the last JSON encoding/decoding, which did not specify JSON_THROW_ON_ERROR.
https://www.php.net/manual/en/function.json-last-error.php

```suggestion
        if ($options & JSON_THROW_ON_ERROR === 0 && json_last_error() !== JSON_ERROR_NONE) {
```

Reply:
Why is the check needed? 
If `JSON_THROW_ON_ERROR` is specified, `json_decode()` throws an Exception already.
If there is an error in decoding, we do not reach this `if` statement.

Reply:
I don't think we need this.

Reply:
`JSON_THROW_ON_ERROR` is used inside `json_(de|en)code`.

How about put the json_decode with JSON_THROW_ON_ERROR inside a try/catch?

Reply:
We can, but why do we want to hide the native exception? If I use `JSON_THROW_ON_ERROR` flag, I expect `JsonException` to be thrown.

Reply:
In this case, a dev sets `JSON_THROW_ON_ERROR` intentionally.
So I also think we don't need to convert `JsonException` to a framework exception.

Reply:
Yes I was wrong: the if statement is only reached when it doesn't throw so the check is unnecessary. My mistake.